lib/hanami/configuration/router.rb

```diff
@@ -1,50 +1,50 @@
 # frozen_string_literal: true
 
+require "dry/configurable"
+require_relative "../application/routing/resolver"
+
 module Hanami
   class Configuration
     # Hanami router configuration
     #
     # @since 2.0.0
     # @api private
     class Router
-      # @api private
-      # @since 2.0.0
-      attr_writer :routes
+      include Dry::Configurable
 
-      # @api private
-      # @since 2.0.0
-      attr_reader :routes
+      attr_reader :base_configuration
+      private :base_configuration
 
       # @api private
       # @since 2.0.0
-      attr_writer :resolver
-
-      # @api private
-      # @since 2.0.0
-      def initialize(base_url, routes: DEFAULT_ROUTES)
-        @base_url = base_url
-        @routes = routes
+      def initialize(base_configuration)
+        # Base configuration is provided so the router can access the `base_url`
+        @base_configuration = base_configuration
       end
 
-      # @api private
-      # @since 2.0.0
-      def resolver
-        @resolver ||= begin
-                        require_relative "../application/routing/resolver"
-                        Application::Routing::Resolver
-                      end
-      end
+      setting :routes_path, "config/routes"
```

Comment:
@timriley Shouldn't this respect `File::SEPARATOR`?

```ruby
setting :routes_path, Hanami.root.join("config", "routes")
```

Not sure if we need to append `.rb` suffix there.

Reply:
@jodosha oh, this is a good reminder! I sometimes write `root.join("foo/bar")` too - totally forgot about the separator thing.

Reply:
Happy to convert this one to join using `File::SEPARATOR`. I think we may need to do a serious sweep across all gems to make sure we're doing it properly across the board.

I wish Ruby's file path joining respected this out of the box 😩

Reply:
Fixed this, and another instance 👍🏼